autoscaler/controllers/gateway/root.go

```diff
@@ -44,21 +47,26 @@ func Sync(ctx context.Context, k8sClient client.Client, scheme *runtime.Scheme,
 		return err
 	}
 	// Add the generic batch processor to the list of processors
-	processors.Items = append(processors.Items, common.GetGenericBatchProcessor())
+	processors.Items = append(processors.Items, commonconf.GetGenericBatchProcessor())
 
 	odigosSystemNamespaceName := env.GetCurrentNamespace()
-	var odigosConfig odigosv1.OdigosConfiguration
-	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: odigosSystemNamespaceName, Name: consts.OdigosConfigurationName}, &odigosConfig); err != nil {
+	var configMap v1.ConfigMap
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: odigosSystemNamespaceName, Name: consts.OdigosConfigurationName}, &configMap); err != nil {
 		logger.Error(err, "failed to get odigos config")
 		return err
 	}
+	var odigosConfig common.OdigosConfiguration
+	if err := yaml.Unmarshal([]byte(configMap.Data[consts.OdigosConfigurationFileName]), &odigosConfig); err != nil {
+		logger.Error(err, "failed to parse odigos config from configmap")
+		return err
+	}
```

Comment:
This looks like a code block we would probably need in multiple places, maybe it's worth extracting it to `k8sutils`?